model/file_info.go

```diff
@@ -170,10 +187,10 @@ func GetInfoForBytes(name string, data []byte) (*FileInfo, *AppError) {
 
 	if info.IsImage() {
 		// Only set the width and height if it's actually an image that we can understand
-		if config, _, err := image.DecodeConfig(bytes.NewReader(data)); err == nil {
-			info.Width = config.Width
-			info.Height = config.Height
-
+		if im, _, err := image.Decode(bytes.NewReader(data)); err == nil {
```

Comment:
This is going to load the whole image into memory earlier in the file upload instead of just loading some of its headers. Is this still happening after the check of the file size/image dimensions? I assume we check the file size in the API call, but I can't remember where we check image dimensions to avoid a zip bomb image being uploaded.

Although since I'm having two thoughts on everything today, maybe we could avoid decoding the image twice somehow since that uses double the memory? Maybe `info.MiniPreview` is not set immediately and we could set it as part of `processImages`, just not asynchronously?

Reply:
Agreed